developers/contracts/xlink-staking.md

```diff
@@ -0,0 +1,458 @@
+
+In summary, these operations are typically accessed by users through intermediary contracts registered as DAO extensions. However, they can also be directly called by a proposal contract executed by the DAO.
+
+#### End user call
+
+```mermaid
+flowchart LR
+    User --Sends transtaction--> i["DAO extension"]
+    i["DAO extension"] -.-> B((Call type))
+    B --contract-call--> sm1[Staking Manager]
+    B --as-contract--> sm2[Staking Manager]
+    sm1 -.-o box1[["**tx-sender**: User
+    **contract-caller**: DAO extension"]]
+    sm2 -.-o box2[["**tx-sender** & **contract-caller**: DAO extension"]]
+```
+
+The `as-contract` call type is the most common approach for liquid staking implementations, where the intermediary contract stakes on behalf of users and provides them with a receipt in the form of a new (rebasing) token.
+
+#### Governance call
+
+```mermaid
+flowchart LR
+    Operator --Sends transtaction--> dao["Executor DAO"]
+    dao --as-contract--> i["Proposal 
+    contract"]
+    i --contract-call--> sm[Staking Manager]
+    sm -.-o box[["**tx-sender**: Executor DAO 
+    **contract-caller**: Proposal contract"]]
+```
+
+### Mechanics
+
+In the Staking Manager, the staker is represented by the `tx-sender`. As explained above, depending on the intermediary contract design, the staker may either be an end user or the endpoint/façade acting on behalf of the user. Therefore, every reference to "staker" (or `user`, as defined within the contract) applies to any of these two possibilities.
+
+Upon staking, the shares corresponding to the amount being staked are calculated and stored in the [`user-shares`](#user-shares) map. This is the staking position and represents the user's portion of the total amount staked. During this operation, the amount to stake is transferred from the `tx-sender` to the Staking Manager, and since the total amount staked and total shares increased, the contract storage is updated to reflect the new state. Refer to the [`stake`](#stake) function for detailed information.
+
+Unstaking performs the inverse operations. It first calculates the shares that correspond to the amount willing to unstake. Then, the unstaked amount is transferred from the Staking Manager to the `tx-sender` and, finally, state upates are performed to decrease user shares, total shares and total amount staked.
+
+{% hint style="info" %}
+Note that both **staking** and **unstaking** operations involve shares management and mutate the staking status of a token in three dimensions: total shares, total amount staked and user's share-based staking position.
+{% endhint %}
+
+#### Amount to shares conversion
```

Comment:
Perhaps I would add "Token" to amount to shares conversion...

#### **Token** amount to shares conversion

Reply:
Entiendo que luego se usa `amount-to-shares` en muchos lugares.. quizas pondria:

Reply:
#### (Token) amount to shares conversion  (`amount-to-shares`)

Reply:
Great. Changed to **Token amount to shares conversion**. Seems OK. The function that is utilized is: `get-shares-given-amount`.